src/server/identity/server/dex_api.go

```diff
@@ -156,18 +167,24 @@ func (a *dexAPI) updateConnector(in *identity.UpdateIDPConnectorRequest) error {
 			c.Name = in.Connector.Name
 		}
 
-		if in.Connector.JsonConfig != "" && in.Connector.JsonConfig != "null" {
-			c.Config = []byte(in.Connector.JsonConfig)
-		}
-
 		if in.Connector.Type != "" {
 			c.Type = in.Connector.Type
 		}
 
+		// dexidp doesn't support unmarshalling from yaml, so we have to convert to json.
+		// If config is already json, then this is a no-op under the hood.
+		config, err := yaml.YAMLToJSON(PickConfig(in.Connector.Config, in.Connector.JsonConfig))
+		if err != nil {
+			return dex_storage.Connector{}, errors.EnsureStack(err)
+		}
+
+		if string(config) != "" && string(config) != "null" {
+			c.Config = config
```

Comment:
otherwise does c.Config == nil ?

Reply:
Yes, I believe so

Reply:
I think it make sense to throw an error in the case that the client passes us an empty config

Reply:
Some of our tests omit config on updates (so they don't need to be aware of the current config). It seems like the intended behavior for those tests is that omitting the config should work on an update, but not on a create of a new connector.